TeensyDmx.cpp

```diff
@@ -1021,14 +1436,52 @@ void TeensyDmx::respondMessage(uint16_t nackReason)
         m_rdmBuffer.dataLength = 2;
         putUInt16(&m_rdmBuffer.data, nackReason);
     }
-    m_rdmBuffer.length = m_rdmBuffer.dataLength + RDM_PACKET_SIZE_NO_PD;  // total packet length
 
     ++m_rdmBuffer.cmdClass;
 
+    sendRDMMessage();
+}
+
+
+void TeensyDmx::buildSendRDMMessage(byte *uid, uint8_t commandClass, uint16_t pid) {
+    if (m_rdm != nullptr) {
+        m_rdmBuffer.startCode = E120_SC_RDM;
+        m_rdmBuffer.subStartCode = E120_SC_SUB_MESSAGE;
+
+        memcpy(m_rdmBuffer.destId, uid, RDM_UID_LENGTH);
+        memcpy(m_rdmBuffer.sourceId, m_rdm->uid, RDM_UID_LENGTH);
+
+        // Sub Dev
+        putUInt16(&m_rdmBuffer.subDev, RDM_ROOT_DEVICE);
+
+        m_rdmBuffer.messageCount = 0; // Number of queued messages
+        m_rdmBuffer.responseType = 1; // Port 1
+        m_rdmBuffer.transNo = 0;
+        // Parameter
+        putUInt16(&m_rdmBuffer.parameter, pid);
+
+        m_rdmBuffer.cmdClass = commandClass;
+
+        sendRDMMessage();
+        m_controllerState = ControllerState::RDM_MESSAGE;
+        // TODO(Peter): Set to RDM_BROADCAST if dest is broadcast
```

Comment:
A waiting TODO.

Reply:
Yep one of many, hence my initial question about how you want to play this; one huge PR, or partial support, or a branch.

Reply:
I suppose it depends on how much of an issue they are.  Does this not limit discovery?  I assume not since you were able to test it?

Reply:
Done this one.

This doesn't limit discovery, although proper discovery within the lib is still on the TODO list. This just impacts on callback behaviour.